### PR TITLE
Make systests work with internal Jenkins CI system.

### LINKED ISSUE #565: Add a Jenkinsfile to help orchestrate internal test jobs.
* Title: Add a Jenkinsfile to help orchestrate internal test jobs.

#### Description
- This is an enhancement for our internal QE processes/tools.
- As part of our move to Jenkins for systest CI, we should add a Jenkinsfile to each project under test.
- This Jenkinsfile will control the "how" and "where" of our systest runs (ie. it defines the build execution steps and specifies the agent/Docker image to use).
- The "what" and "when" (ie. what jobs to run and when to run them) will be controlled by the Jenkins server (ie. configured via the Jenkins GUI).

### PR DESCRIPTION
**Pull request by larkinkevin**
@pjbreaux, @mattgreene

#### What issues does this address?
Fixes #565

#### What's this change do?
Adds a Jenkinsfile to facilitate our move to Jenkins for internal systest CI.

#### Where should the reviewer start?
n/a

#### Any background context?
no

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent { docker defaultWorker.getConfig("openstack-test") }
+    options {
+        ansiColor('xterm')
+        timestamps()
+        timeout(time: 2, unit: "HOURS")
```
Comment (pjbreaux) [inline Jenkinsfile]: If this is the timeout for one target, as of now, we could reasonably hit this timeout by adding a dozen end-to-end tests. I say we bump to 2.5 hours.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        stage("systest") {
+            steps {
+                sh '''
+                    # - initialize job control vars
+                    # - JOB_NAME is an environment variable that is
+                    #   automatically provided by Jenkins
+                    #   (eg. "openstack/driver/mitaka/12.1.1-overcloud-vxlan")
+                    export PROGRAM=$(echo $JOB_NAME | cut -d "/" -f 1)
+                    export PROJECT=$(echo $JOB_NAME | cut -d "/" -f 2)
+                    export BRANCH=$(echo $JOB_NAME | cut -d "/" -f 3)
```
Comment (pjbreaux) [inline Jenkinsfile]: In the near future, our branches will contain forward slashes (stable/newton). This cut would not capture that branch name in full.

Comment (larkinkevin) [inline Jenkinsfile]: Jenkins job names cannot contain forward slashes (Jenkins won't let you create them).  So this cut should work as expected.

When you add a Jenkins folder to track the jobs for "stable/newton" you'll have to pick Jenkins-frienly name (like "stable-newton" or "stable_newton").

File: Jenkinsfile (hunk 1 above)
Comment (larkinkevin) [inline Jenkinsfile]: I can certainly bump the timeout if you want.  The current timeout of 2 hours is based on the current work load (where each job tends to complete in 1-1.5 hours).

File: systest/Makefile [hunk 3]
```diff
@@ -35,7 +35,7 @@ export EXCLUDE_DIR := $(MAKEFILE_DIR)/exclude/
 
 # Repos that are used for the tests and environment
 # TLC
-export TLC_DIR := /home/buildbot/tlc
+export TLC_DIR := /home/jenkins/tlc
```
Comment (pjbreaux) [inline systest/Makefile]: Did you grep the systest directory for all occurrences of 'buildbot' in a path?

Comment (larkinkevin) [inline systest/Makefile]: Yep.

**Comment by mattgreene** [comment]
Holding off on the merge until we clear the next nightly run.

**Pull request merged**